src/test/java/org/openrewrite/reactive/reactor/ReactorDoAfterSuccessOrErrorToTapTest.java

```diff
@@ -0,0 +1,435 @@
+                }
+            }
+            """,
+            """
+            import reactor.core.observability.DefaultSignalListener;
+            import reactor.core.publisher.Mono;
+            import reactor.core.publisher.SignalType;
+
+            class SomeClass {
+                void doSomething(Mono<String> mono) {
+                    mono.tap(() -> new DefaultSignalListener<>() {
+                        @Override
+                        public void doFinally(SignalType terminationType) {
+                            System.out.println("other logs");
+                        }
+
+                        @Override
+                        public void doOnNext(String result) {
+                            System.out.println("success" + result);
+                        }
+
+                        @Override
+                        public void doOnError(Throwable error) {
+                            System.out.println("error" + error);
+                        }
+                    }).subscribe();
+                }
+            }
+            """
+          )
+        );
+    }
+
+    @Test
+    void emptyLambda() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+            import reactor.core.publisher.Mono;
+
+            class SomeClass {
+                void doSomething(Mono<String> mono) {
+                    mono.doAfterSuccessOrError((result, error) -> {
+                    }).subscribe();
+                }
+            }
+            """,
+            """
+            import reactor.core.observability.DefaultSignalListener;
+            import reactor.core.publisher.Mono;
+            import reactor.core.publisher.SignalType;
+
+            class SomeClass {
+                void doSomething(Mono<String> mono) {
+                    mono.tap(() -> new DefaultSignalListener<>() {
+                        @Override
+                        public void doFinally(SignalType terminationType) {
+                        }
+
+                        @Override
+                        public void doOnNext(String result) {
+                        }
+
+                        @Override
+                        public void doOnError(Throwable error) {
+                        }
+                    }).subscribe();
+                }
+            }
+            """
+          )
+        );
+    }
+
+    @Test
+    void refactorRandomStatementOrder() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+            import reactor.core.publisher.Mono;
+
+            class SomeClass {
+                void doSomething(Mono<String> mono) {
+                    mono.doAfterSuccessOrError((result, error) -> {
+                        doSomething();
+                        if (error != null) {
+                            System.out.println("error" + error);
+                            doSomething(error);
+                        } else {
```

Comment:
If I add a `return` right here, then I end up with an after block with unreachable code. I'd think a return is pretty common, and we should account for that in our logic and not change what code runs, or make no change at all.
```suggestion
                            doSomething(error);
                            return;
                        } else {
```

Reply:
Fixed it with latest version.